third_party/terraform/resources/resource_google_service_account.go

```diff
@@ -121,31 +123,33 @@ func resourceGoogleServiceAccountDelete(d *schema.ResourceData, meta interface{}
 
 func resourceGoogleServiceAccountUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	sa, err := config.clientIAM.Projects.ServiceAccounts.Get(d.Id()).Do()
-	if err != nil {
-		return fmt.Errorf("Error retrieving service account %q: %s", d.Id(), err)
-	}
 	updateMask := make([]string, 0)
 	if d.HasChange("description") {
 		updateMask = append(updateMask, "description")
 	}
 	if d.HasChange("display_name") {
 		updateMask = append(updateMask, "display_name")
 	}
-	_, err = config.clientIAM.Projects.ServiceAccounts.Patch(d.Id(),
-		&iam.PatchServiceAccountRequest{
-			UpdateMask: strings.Join(updateMask, ","),
-			ServiceAccount: &iam.ServiceAccount{
-				DisplayName: d.Get("display_name").(string),
-				Description: d.Get("description").(string),
-				Etag:        sa.Etag,
-			},
-		}).Do()
-	if err != nil {
-		return err
+	if len(updateMask) > 0 {
```

Comment:
What's the motivation behind these changes to `..Update`? I'm not sure the Terraform SDK would call into `..Update` without a diff.

Reply:
Ah... this happened during merging #2650 by @ndmckinley into my original code 81c263d94bc541c8b01a8fe32ce97dcbdd8e376a, but well you are right, we don't need this.

Reply:
fixed ea0839f